cmd/kubectl-k8ssandra/register/register_test.go

```diff
@@ -59,55 +70,28 @@ func TestRegister(t *testing.T) {
 		SourceNamespace:  "source-namespace",
 		DestNamespace:    "dest-namespace",
 		ServiceAccount:   "k8ssandra-operator",
-		Context:          context.TODO(),
+		Context:          ctx,
 		DestinationName:  "test-destination",
 	}
-	ctx := context.Background()
-
+	// Continue reconciliation
 	require.Eventually(func() bool {
-		if err := ex.RegisterCluster(); err != nil {
-			if errors.Is(err, NonRecoverableError{}) {
-				require.FailNow(fmt.Sprintf("Registration failed: %s", err.Error()))
-			}
-			if err.Error() == "no secret found for service account k8ssandra-operator" {
-				return true
-			}
-			return false
-		}
-		return true
-	}, time.Second*5, time.Millisecond*100)
-
-	// This relies on a controller that is not running in the envtest.
+		res := ex.RegisterCluster()
+		return res == nil
+	}, time.Second*300, time.Second*1)
 
-	desiredSaSecret := &corev1.Secret{}
-	require.NoError(client1.Get(context.Background(), client.ObjectKey{Name: "k8ssandra-operator-secret", Namespace: "source-namespace"}, desiredSaSecret))
-	patch := client.MergeFrom(desiredSaSecret.DeepCopy())
-	desiredSaSecret.Data = map[string][]byte{
-		"token":  []byte("test-token"),
-		"ca.crt": []byte("test-ca"),
-	}
-	require.NoError(client1.Patch(ctx, desiredSaSecret, patch))
+	sourceSecret := &corev1.Secret{}
+	// Ensure secret created.
+	require.Eventually(func() bool {
+		err := client1.Get(ctx, types.NamespacedName{Name: "k8ssandra-operator-secret", Namespace: "source-namespace"}, sourceSecret)
+		return err == nil
+	}, time.Second*60, time.Second*5)
```

Comment:
issue: Reduce the polling interval (and the maximum time so that we bailout faster). I reduced these already to speed up the test from 16s to 2.4s and I'd hate to see the test taking more time than necessary.